File: gdx2d-demoDesktop/src/main/kotlin/ch/hevs/gdx2d/demos/shaders/advanced/DemoMultipleTextures.kt
package ch.hevs.gdx2d.demos.shaders.advanced

import ch.hevs.gdx2d.desktop.PortableApplication
import ch.hevs.gdx2d.lib.GdxGraphics
import com.badlogic.gdx.Gdx

/**
 * Demonstrates the use of multiple textures in a shader
 *
 * @author Pierre-André Mudry (mui)
 * @version 0.5
 */
class DemoMultipleTextures : PortableApplication() {

    var time = 0f
    var i = 0

    override fun onInit() {
        this.setTitle("Multiple textures passing to shader, mui 2013")
    }

    override fun onGraphicRender(g: GdxGraphics) {
        if (g.shaderRenderer == null) {
            g.setShader("shader/advanced/multiple_textures.fp")
            g.shaderRenderer.addTexture("images/lena.png", "texture0")
            g.shaderRenderer.addTexture("images/mandrill.jpg", "texture1")
        }

        g.shaderRenderer.setUniform("textureChosen", i)

        g.clear()
        time += Gdx.graphics.deltaTime
        g.drawShader(time)
        g.drawFPS()
        g.drawSchoolLogo()
    }

    override fun onClick(x: Int, y: Int, button: Int) {
        super.onClick(x, y, button)
        i = (i + 1) % 2
    }
}

fun main(args: Array<String>) {
  DemoMultipleTextures()
}

File: gdx2d-demoDesktop/src/main/scala/ch/hevs/gdx2d/demos/shaders/advanced/DemoMultipleTextures.scala
package ch.hevs.gdx2d.demos.shaders.advanced

import ch.hevs.gdx2d.desktop.DesktopApplication
import ch.hevs.gdx2d.lib.GdxGraphics
import com.badlogic.gdx.Gdx

/**
 * Demonstrates the use of multiple textures in a shader.
 *
 * @author Pierre-André Mudry (mui)
 * @version 2.0
 */
class DemoMultipleTextures extends DesktopApplication {

  private var time = 0f
  private var selectedTexture = 0

  override def onInit(): Unit = {
    setTitle("Multiple textures passing to shader, mui 2013")
  }

  override def onGraphicRender(g: GdxGraphics): Unit = {
    if (g.getShaderRenderer == null) {
      g.setShader("shader/advanced/multiple_textures.fp")
      g.getShaderRenderer.addTexture("images/lena.png", "texture0")
      g.getShaderRenderer.addTexture("images/mandrill.jpg", "texture1")
    }

    g.getShaderRenderer.setUniform("textureChosen", selectedTexture)

    g.clear()
    time += Gdx.graphics.getDeltaTime
    g.drawShader(time)
    g.drawFPS()
    g.drawSchoolLogo()
  }

  override def onClick(x: Int, y: Int, button: Int): Unit = {
    super.onClick(x, y, button)
    selectedTexture = (selectedTexture + 1) % 2
  }
}

object DemoMultipleTextures {
  def main(args: Array[String]): Unit = {
    new DemoMultipleTextures().launch()
  }
}
